spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClientAutoConfiguration.java

```diff
@@ -90,12 +94,24 @@ public KubernetesDiscoveryClientHealthIndicatorInitializer indicatorInitializer(
 
 	@Bean
 	@ConditionalOnMissingBean
-	public KubernetesInformerDiscoveryClient kubernetesClientInformerDiscoveryClient(
```

Comment:
depending on selective namespaces being enabled or not, create the appropriate client. One takes a List of various things, one single items.